source/schemas/types.txt

```diff
@@ -509,3 +509,80 @@ with the ``title`` field set to the embedded object's type name:
             }
       }
    }
+
+.. _geospatial:
+
+Geospatial Data
+---------------
+
+Geospatial data describes points and other data on the earth's surface. App 
+Services supports defining GeoJSON Point data in a schema. 
+
+GeoJSON Point
+~~~~~~~~~~~~~
+
+A GeoJSON Point is a single location on the Earth's surface, described by its 
+longitude and latitude. To ensure wide compatibility, App Services stores these 
+points in accordance to the `GeoJSON spec <https://datatracker.ietf.org/doc/html/rfc7946>`_.
+To store a GeoPoint in Atlas, you must follow these rules:
+
+- It must be of type ``object``
+- It must be embedded within another type
+- It must contain the following fields:
+
+  - ``type``, which is always set to "Point"
+  - ``coordinates``, of type ``array``. The ``coordinates`` array contains 
+    exactly 2 double values. The first double is the longitude of the point and 
+    the second is the latitude.
+
+- Additional properties are allowed.
+ 
+The following is the schema of an object where the "location" property is 
+a GeoPoint:
+
+.. code-block:: json
+
+   {
+     "title": "MyObject",
+     "properties": {
+       "location": {
+         "bsonType": "object",
+         "required": [ "type" ],
+         "properties": {
+           "type": {
+             "bsonType": "string",
+             "const": "Point"
+           },
+           "coordinates": {
+             "bsonType": "array",
+             "items": {
+               "bsonType": "double",
+                "minItems": 2,
```

Comment:
[nit] We actually decided that "minItems" and "maxItems" weren't worth adding into the JSON schema definition, and the only requirement for "coordinates" is that it must have "bsonType": "array", and items "bsonType": "double".

Reply:
@jonatlee  So the schema doesn't enforce the presence of the "type" property or it's value as "Point"?

Reply:
"type" is required -- it needs to be explicitly included in the required fields list in order for the embedded object to be treated as a Point by the server. 

Outside of any schema validation, the value of "type" is checked by the server on properties that have been marked as a GeoPoint as writes are coming in to ensure that it never gets set to anything other than "Point".

If "type" gets set to something other than "Point" by a mobile device, the server will issue a compensating write to the client which effectively reverts the write. 

If "type" gets set to something other than "Point" due to a change directly to MongoDB, the server will mark the written document as unsyncable.